server/server.py

```diff
@@ -18,139 +18,212 @@
-#                 cursor.execute("INSERT INTO adminTokens VALUES(%s,%s)", (adminToken, expirationTime))
-#                 resp.set_cookie('adminToken', adminToken)
-#             con.commit()
-#             # Save token as cookie
-#             resp.set_cookie('token', token)
-#             # Remove the admin redirect token if needed
-#             resp.set_cookie('adminRefresh', '', expires=0)
-#             return resp
-
-
+    return render_template('home.html', username=current_user.full_name)
+
+@app.route("/admin")
+@admin_required
+def admin_dashboard():
+    return render_template("admin.html")
+
+@app.route("/superadmin")
+@super_admin_required
+def superadmin_dashboard():
+    return jsonify(msg="🚀 Welcome, superadmin!"), 200
+
+@app.route("/logout")
+def logout():
+    next_url = request.args.get("next") or url_for("stickerMap")
+    resp = flask.make_response(redirect(next_url))
+    resp.delete_cookie("access_token_cookie", path="/")
+    return resp
+
+# ─── OpenID /auth Flow ───────────────────────────────────────────────────
+@app.route("/login")
+def login():
+    code = request.args.get("code")
+    if not code:
+        next_url = request.args.get("next") or url_for("stickerMap")
+        auth_url = (
+            f"{Config.OIDC_ISSUER_BASE}/api/oauth/authorize"
+            f"?client_id={Config.OIDC_CLIENT_ID}"
+            f"&scope={Config.OIDC_SCOPES}"
+            f"&redirect_uri={url_for('login', _external=True)}"
+            f"&response_type=code"
+            f"&state={next_url}"
+        )
+        return render_template("login.html", loginUrl=auth_url)
+
+    next_url = request.args.get("next") or url_for("stickerMap")
+
+    # Exchange code for tokens
+    token_response = requests.post(
+        f"{Config.OIDC_ISSUER_BASE}/api/oauth/token",
+        data={
+            "grant_type":   "authorization_code",
+            "code":         code,
+            "redirect_uri": url_for('login', _external=True)
+        },
+        auth=(Config.OIDC_CLIENT_ID, Config.OIDC_CLIENT_SECRET)
+    ).json()
+
+    # Redirect user to login when there is no id_token
+    id_token = token_response.get("id_token")
+    if not id_token:
+        return redirect(url_for("login"))
+
+    next_url = request.args.get("state") or url_for("stickerMap")
+
+    # Set the Koala ID‑token as the cookie
+    resp = flask.make_response(redirect(next_url))
+    resp.set_cookie(
+        "access_token_cookie",
+        id_token,
+        httponly=True,
+        secure=Config.JWT_COOKIE_SECURE,
+        path="/"
+    )
+    return resp
+
+# ─── Backend routes ───────────────────────────────────────────────────
 @app.route('/upload', methods=['GET', 'POST'])
 def uploadSticker():
-    # Check token if required
-    if os.getenv('STICKER_MAP_REQUIRE_LOGIN') == "True":
-        if not checkToken(request.cookies.get('token')):
-            return json.dumps({'status': '403', 'error': 'Not authenticated or cookies disabled.'}), 405
```

Comment:
You removed this check, but shouldn't a `@require_login` or something be added instead? Same for other methods, but I will repeat my self there.

Reply:
Sure we can! Next session, I'm quite busy the next 2 weeks. Or perhaps someone else, I know Sam and Job know OAuth.